server/jetstream_api.go

```diff
@@ -3673,13 +3676,13 @@ func (s *Server) streamSnapshot(ci *ClientInfo, acc *Account, mset *stream, sr *
 			}
 		}
 		ackReply := fmt.Sprintf("%s.%d.%d", ackSubj, len(chunk), index)
-		mset.outq.send(newJSPubMsg(reply, _EMPTY_, ackReply, nil, chunk, nil, 0))
+		mset.outq.send(newJSPubMsg(reply, _EMPTY_, ackReply, hdr, chunk, nil, 0))
 		atomic.AddInt32(&out, int32(len(chunk)))
 	}
 done:
 	// Send last EOF
-	// TODO(dlc) - place hash in header
-	mset.outq.send(newJSPubMsg(reply, _EMPTY_, _EMPTY_, nil, nil, nil, 0))
+	hdr := genHeader(nil, "Nats-Snapshot-Checksum-Sha256", hex.EncodeToString(hash.Sum(nil)))
```

Comment:
Make a constant and probably just say checksum and value will have type, e.g. "sha256: XXXXX"